x-pack/plugins/ingest_manager/server/routes/agent_config/handlers.ts

```diff
@@ -229,3 +230,37 @@ export const getFullAgentConfig: RequestHandler<TypeOf<
     });
   }
 };
+
+export const downloadFullAgentConfig: RequestHandler<TypeOf<
+  typeof GetFullAgentConfigRequestSchema.params
+>> = async (context, request, response) => {
+  const soClient = context.core.savedObjects.client;
+  const {
+    params: { agentConfigId },
+  } = request;
+
+  try {
+    const fullAgentConfig = await agentConfigService.getFullConfig(soClient, agentConfigId);
+    if (fullAgentConfig) {
+      const body = configToYaml(fullAgentConfig);
+      const headers: ResponseHeaders = {
+        'content-type': 'text/x-yaml',
+        'content-disposition': `attachment; filename="elastic-agent-config-${fullAgentConfig.id}.yml"`,
```

Comment:
@ruflin @ph I'm defaulting the downloaded YAML file name to `elastic-agent-config-{agentConfigId}.yml` but feel that it may be a bit long. Is there another convention we prefer?

Reply:
The default config that is picked up by the agent is called `elastic-agent.yml`: https://github.com/elastic/beats/blob/master/x-pack/elastic-agent/elastic-agent.yml Taking this name would make it very easy to copy / paste. Having the id inside makes it nice for tracking but not sure we need it at the moment?